p2panda-js/src/types.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Arguments for publishing the next entry
+ */
+export type EntryArgs = {
+  entryHashSkiplink: string | undefined;
+  entryHashBacklink: string | undefined;
+  lastSeqNum: number | undefined;
+  logId: number;
+};
+
+/**
+ * Entry record received from aquadoggo
+ */
+export type EncodedEntry = {
+  author: string;
+  entryBytes: string;
+  entryHash: string;
+  logId: number;
+  payloadBytes: string;
+  payloadHash: string;
+  seqNum: number;
+};
+
+/**
+ * Entry record from aquadoggo with decoded `Entry`
+ */
+export type EntryRecord = Entry & {
+  encoded: EncodedEntry;
+};
+
+/**
+ * Decoded entry containing optional `Message`
+ */
+export type Entry = {
+  entryHashBacklink: string | null;
+  entryHashSkiplink: string | null;
+  logId: number;
+  message: Message | null;
+  seqNum: number;
+};
+
+/**
+ * Decoded form of a message, which can create, update or delete instances
+ */
+export type Message = {
+  action: 'create' | 'update' | 'delete';
+  schema: string;
+  fields: Fields;
+};
+
+/**
+ * Object containing message field values
+ */
+export type Fields = {
+  // currently only a schema with a text message is supported
+  // [fieldname: string]: boolean | number | string;
+  [fieldname: string]: string;
+};
+
+/**
+ * Decoded entry containing optional `Message`
+ */
+export type EntryTagged = {
+  entryHashBacklink: string | null;
+  entryHashSkiplink: string | null;
+  logId: number;
+  message: MessageTagged | null;
+  seqNum: number;
+};
+
+/**
+ * Decoded form of a message, which can create, update or delete instances
+ */
+export type MessageTagged = {
+  action: 'create' | 'update' | 'delete';
+  schema: string;
+  fields: FieldsTagged;
+};
+
+/**
+ * Object containing message fields in tagged form
+ */
+export type FieldsTagged = {
+  // currently only a schema with a text message is supported
```

Comment:
Is this true? I thought we've added support for other types as well now :+1:

Reply:
See: https://github.com/p2panda/p2panda/blob/main/p2panda-rs/src/wasm.rs#L70

Reply:
We're limited in what we can do wrt field types until we can retrieve user schemas i think. So in terms of this PR we should just change the comment. Or did you have something more in mind?

Reply:
Change the comment, and raise an issue, of course 😉

Reply:
We could try and infer the schema from the data message fields that users supply us with. Then we could use more than just `string`  fields. Would we get a good error message if the wrong message field data is supplied?

Reply:
Ok, I have added support for `int` and `boolean`! `relationship` and `float` will have to wait until we have better schema support :)